spec/locker/advisory_spec.rb

```diff
@@ -72,6 +72,25 @@
       expect(lock1_result).to be true
       expect(lock2_result).to be true
     end
+
+    it "blocks with a timeout" do
+      started_at = Time.now
+      t1 = Thread.new do
+        Locker::Advisory.run("foo") do
+          sleep 2
+        end
+      end
+
+      t2 = Thread.new do
+        sleep 0.5
+        Locker::Advisory.run("foo", :blocking => true, :block_timeout => 1){}
```

Comment:
Can you check the return value on this as part of the test, too?  (Since that's part of the normal/expected usage.)